File: bit_manipulation/set_kth_bit/set_kth_bit.cpp
#include<iostream>
using namespace std

int set_kth_bit(int N,int k)
{
        int a;
        a=1<<(k-1)        //1 is right-shifted (k-1) times to set the kth bit '1', while the remaining bits remain unset. 
        N=N|a;            // Using the bitwise OR operator the kth bit of 'N' is set.
        return N;
}
int main()
{
        int N,k;
        cout<<"ENTER THE NUMBER";
        cin>>N;
        cout<<"ENTER THE VALUE OF k";
        cin>>k;

        N=set_kth_bit(N);     /*function to compute the new value after setting the kth bit in binary representation of given number 'N'. 
                                Store the result back in 'N'.*/

        cout<<"After seting the bit the number becomes"<<N;
        return 0;
}

File: bit_manipulation/set_kth_bit/SetKthBit.java
// Java implementation to set the kth bit 

// of the given number 

public class SetKthBit { 

	// function to set the kth bit 

public static int setKthBit(int n, int k) {



	// kth bit of n is being set by this operation 

	return ((1 << k) | n); 

} 

// Driver code 

public static void main(String arg[]) {



	int n = 10, k = 2; 

	System.out.print("Kth bit set number = " + 

							setKthBit(n, k)); 

} 

} 

